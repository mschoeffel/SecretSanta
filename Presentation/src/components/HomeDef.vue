<template>
      <v-container
        class="fill-height"
        fluid
      >
        <v-row
          align="center"
          justify="center"
        >
          <v-col
            cols="12"
            sm="8"
            md="4"
          >
            <v-card class="elevation-12">
              <v-toolbar
                color="primary"
                dark
                flat
              >
                <v-toolbar-title>{{ $t('home.formtitle')}}</v-toolbar-title>
              </v-toolbar>
              <v-card-text>
                <v-form>
                  <v-text-field
                    :label="$t('home.group')"
                    name="group"
                    prepend-icon="mdi-account-group"
                    type="text"
                  />

                  <v-text-field
                    :label="$t('home.name')"
                    name="name"
                    prepend-icon="mdi-account"
                    type="text"
                  />

                  <v-text-field
                    id="password"
                    :label="$t('home.key')"
                    name="key"
                    prepend-icon="mdi-key"
                    type="password"
                  />
                </v-form>
              </v-card-text>
              <v-card-actions>
                <v-spacer />
                <v-btn color="primary">{{ $t('home.draw')}}</v-btn>
              </v-card-actions>
            </v-card>
          </v-col>
        </v-row>
      </v-container>
</template>

<script>
  export default {
    props: {
      source: String,
    },
  }
</script>