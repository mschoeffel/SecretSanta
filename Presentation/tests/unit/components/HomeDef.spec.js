import Vue from 'vue'
import Vuetify from 'vuetify'
import VueI18n from 'vue-i18n'

import HomeDef from '@/components/HomeDef'

import { mount, createLocalVue } from '@vue/test-utils';

const localVue = createLocalVue()

Vue.use(VueI18n)
const i18n = new VueI18n({});

describe('HomeDef.vue', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const mountFunction = options => {
    return mount(HomeDef, {
      localVue,
      vuetify,
      i18n,
      ...options,
    })
  }

  it('should render correct home form', () => {
    const wrapper = mountFunction()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should give invalid input to empty form', () =>{
    const wrapper = mountFunction()

    wrapper.find('v-btn[id=next]').trigger('click')

    expect(wrapper.vm.step).toBe(1)
    expect(wrapper.vm.groupnameerror).toBe("home.validation-groupnamerequired")
    expect(wrapper.vm.membernameerror).toBe("home.validation-membernamerequired")
    expect(wrapper.vm.keyerror).toBe("home.validation-keyrequired")
    expect(wrapper.vm.displayerror).toBe("home.validationerror")
  })
  
  it('should give invalid input to empty groupname', () =>{
    const wrapper = mountFunction()

    wrapper.setData({
        groupname: "",
        name: "abcdef",
        key: "123456"
    })
    wrapper.find('v-btn[id=next]').trigger('click')

    expect(wrapper.vm.step).toBe(1)
    expect(wrapper.vm.groupnameerror).toBe("home.validation-groupnamerequired")
    expect(wrapper.vm.membernameerror).toBe("")
    expect(wrapper.vm.keyerror).toBe("")
    expect(wrapper.vm.displayerror).toBe("home.validationerror")
  })

})
