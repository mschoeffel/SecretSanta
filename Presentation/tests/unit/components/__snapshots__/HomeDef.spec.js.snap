// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeDef.vue should render correct home form 1`] = `
<v-container fluid="" class="fill-height">
  <v-row align="center" justify="center">
    <v-col cols="12" sm="12" md="8">
      <v-card class="elevation-12">
        <v-toolbar color="primary" dark="" flat="">
          <v-toolbar-title name="title" id="title">home.formtitle</v-toolbar-title>
        </v-toolbar>
        <v-card-text>
          <v-stepper alt-labels="">
            <v-stepper-header>
              <v-stepper-step step="1">home.input</v-stepper-step>
              <v-divider></v-divider>
              <v-stepper-step step="2">home.result</v-stepper-step>
            </v-stepper-header>
            <v-stepper-items>
              <v-stepper-content step="1">
                <!---->
                <v-text-field label="home.group" name="groupname" id="groupname" type="text" prepend-icon="mdi-account-group" counter="15" error-messages=""></v-text-field>
                <v-text-field label="home.name" name="name" id="name" type="text" prepend-icon="mdi-account" counter="15" error-messages=""></v-text-field>
                <v-text-field label="home.key" name="key" id="key" type="password" prepend-icon="mdi-key" append-icon="mdi-eye-off" error-messages=""></v-text-field>
              </v-stepper-content>
              <v-stepper-content step="2">
                <!---->
                <p class="font-weight-medium text-center">home.message</p> <br>
                <p class="font-weight-bold text-center"></p>
                <div><br>
                  <p>home.autoaccept</p>
                </div>
              </v-stepper-content>
            </v-stepper-items>
          </v-stepper>
        </v-card-text>
        <v-card-actions>
          <v-btn name="routecreategroup" id="routecreategroup" color="secondary">home.create-group</v-btn>
          <!---->
          <v-spacer></v-spacer>
          <v-btn name="next" id="next" color="primary">home.draw</v-btn>
          <!---->
          <!---->
          <!---->
        </v-card-actions>
      </v-card>
    </v-col>
  </v-row>
</v-container>
`;
